components/public-api/gitpod/experimental/v1/workspaces.proto

```diff
@@ -249,12 +251,37 @@ message WorkspaceInstanceStatus {
     // Admission describes who can access a workspace instance and its ports.
     AdmissionLevel admission = 6;
 
+    // ports is the list of served ports in the workspace.
+    repeated Port ports = 7;
+
     // repo details the Git working copy status of the workspace.
     // Note: this is a best-effort field and more often than not will not be present. Its absence does not
     // indicate the absence of a working copy.
     // contentservice.GitStatus repo = 7;
 }
 
+// PortVisibility defines who may access a workspace port which is guarded by an authentication in the proxy
+enum PortVisibility {
```

Comment:
```suggestion
enum PortPolicy {
```
As it feels more appropriate

Reply:
Another alternative can be `PortPrivacy`. I think it is what VS Code apis are using, i.e. public, private, team, org and so on